spark/dl/src/main/scala/com/intel/analytics/bigdl/example/modeludf/TextClassifier.scala

```diff
@@ -0,0 +1,265 @@
+/*
+ * Copyright 2016 The BigDL Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.intel.analytics.bigdl.example.modeludf
+
+import java.io.File
+import java.util
+
+import com.intel.analytics.bigdl._
+import com.intel.analytics.bigdl.dataset._
+import com.intel.analytics.bigdl.example.textclassification.SimpleTokenizer._
+import com.intel.analytics.bigdl.example.textclassification.{SimpleTokenizer, WordMeta}
+import com.intel.analytics.bigdl.example.modeludf.Options.TextClassificationParams
+import com.intel.analytics.bigdl.nn.{ClassNLLCriterion, _}
+import com.intel.analytics.bigdl.optim._
+import com.intel.analytics.bigdl.tensor.Tensor
+import com.intel.analytics.bigdl.utils.T
+import org.apache.spark.SparkContext
+import org.apache.spark.rdd.RDD
+import org.slf4j.{Logger, LoggerFactory}
+
+import scala.collection.mutable.{ArrayBuffer, Map => MMap}
+import scala.io.Source
+import scala.language.existentials
+
+
+  /**
+  * This example use a (pre-trained GloVe embedding) to convert word to vector,
+  * and uses it to train a text classification model on the 20 Newsgroup dataset
+  * with 20 different categories. This model can achieve around 90% accuracy after
+  * 2 epoches training.
+  */
+class TextClassifier(param: TextClassificationParams) extends Serializable {
```

Comment:
Maybe we should refactor and reuse the previous textclassifier instead of create a new one otherwise there would be many duplicate code and not easy to maintenance.